packages/usa-modal/src/index.js

```diff
@@ -170,104 +171,142 @@ function toggleModal(event) {
       `data-original-${attribute.name}`,
       attribute.value
     );
-  }
+  });
 
-  modalContent.after(originalLocationPlaceHolder);
+  return originalLocationPlaceHolder;
+};
 
-  // Rebuild the modal element
-  modalContent.parentNode.insertBefore(modalWrapper, modalContent);
-  modalWrapper.appendChild(modalContent);
-  modalContent.parentNode.insertBefore(overlayDiv, modalContent);
-  overlayDiv.appendChild(modalContent);
+const setModalAttributes = (baseComponent, targetWrapper) => {
+  const modalID = baseComponent.getAttribute("id");
+  const ariaLabelledBy = baseComponent.getAttribute("aria-labelledby");
+  const ariaDescribedBy = baseComponent.getAttribute("aria-describedby");
+  const forceUserAction = baseComponent.hasAttribute(FORCE_ACTION_ATTRIBUTE);
 
-  // Add classes and attributes
-  modalWrapper.classList.add(HIDDEN_CLASS);
-  modalWrapper.classList.add(WRAPPER_CLASSNAME);
-  overlayDiv.classList.add(OVERLAY_CLASSNAME);
+  if (!ariaLabelledBy)
+    throw new Error(`${modalID} is missing aria-labelledby attribute`);
+
+  if (!ariaDescribedBy)
+    throw new Error(`${modalID} is missing aria-desribedby attribute`);
 
   // Set attributes
-  modalWrapper.setAttribute("role", "dialog");
-  modalWrapper.setAttribute("id", modalID);
+  targetWrapper.setAttribute("role", "dialog");
+  targetWrapper.setAttribute("id", modalID);
 
   if (ariaLabelledBy) {
-    modalWrapper.setAttribute("aria-labelledby", ariaLabelledBy);
+    targetWrapper.setAttribute("aria-labelledby", ariaLabelledBy);
   }
 
   if (ariaDescribedBy) {
-    modalWrapper.setAttribute("aria-describedby", ariaDescribedBy);
+    targetWrapper.setAttribute("aria-describedby", ariaDescribedBy);
   }
 
   if (forceUserAction) {
-    modalWrapper.setAttribute(FORCE_ACTION_ATTRIBUTE, "true");
+    targetWrapper.setAttribute(FORCE_ACTION_ATTRIBUTE, forceUserAction);
   }
 
+  // Add aria-controls
+  const modalClosers = targetWrapper.querySelectorAll(CLOSERS);
+  modalClosers.forEach((el) => {
+    el.setAttribute("aria-controls", modalID);
+  });
+
   // Update the base element HTML
   baseComponent.removeAttribute("id");
   baseComponent.removeAttribute("aria-labelledby");
   baseComponent.removeAttribute("aria-describedby");
   baseComponent.setAttribute("tabindex", "-1");
 
-  // Add aria-controls
-  const modalClosers = modalWrapper.querySelectorAll(CLOSERS);
-  modalClosers.forEach((el) => {
-    el.setAttribute("aria-controls", modalID);
-  });
+  return targetWrapper;
+};
+
+const rebuildModal = (baseComponent) => {
+  const modalContent = baseComponent;
+  const modalWrapper = document.createElement("div");
+  const overlayDiv = document.createElement("div");
+
+  // ? Seperate wrapper and modal?
+  // Rebuild the modal element
+  modalContent.parentNode.insertBefore(modalWrapper, modalContent);
+  modalWrapper.appendChild(modalContent);
+  modalContent.parentNode.insertBefore(overlayDiv, modalContent);
+  overlayDiv.appendChild(modalContent);
```

Comment:
In this specific example, we need the parent `overlay` div on the wrapper so that `setModalAttributes()` can properly set up the modal's closers. 

So the process looks a little more like:
1. Create elements
2. Add classes
3. Structure overlay, wrapper, and content in correct order
4. Set attributes
5. Append fully scaffolded HTML to DOM

Take a look at this updated code and let me know what you think!

Updated in [270c293](https://github.com/uswds/uswds/pull/5315/commits/270c293dce821dca2d79ac92dff84fbac1bd223d)